storage/libradb/src/lib.rs

```diff
@@ -136,10 +136,7 @@ impl LibraDB {
 
         let path = db_root_path.as_ref().join("libradb");
         let instant = Instant::now();
-        let db = Arc::new(
-            DB::open(path.clone(), cf_opts_map)
-                .unwrap_or_else(|e| unrecoverable!("LibraDB open failed: {:?}", e)),
-        );
+        let db = Arc::new(DB::open(path.clone(), cf_opts_map).expect("LibraDB open failed"));
```

Comment:
why leverage expected versus unrecoverable? Do we still need unrecoverable?

Reply:
So unrecoverable was added to replace the panic macro so that MIRAI would be happy. that macro really should live in the mirai-annotations crate but it doesn't currently live there and instead lives in the failure crate I'm trying to remove. So i moved it to another crate. During this process I also found that a number of these (actually all of them) instances could be rewritten using expect (making them shorter and IMO clearer) or context + unwrap. I ended up leaving the macro around though (despite it being currently unused) in case we have need of it again in the future.